### ISSUE TITLE
Profiling a function from Dynamic Loaded library

### ISSUE THREAD

**Issue by ValouBambou** [intel/ittapi#89]
Os/Distro: Archlinux and vtune installed with the intel-oneapi-basekit package (2023 0.0.25537-3)

Context: I want to debug the performance of a specialized generated C code that do some kind of matrix multiplication computation. The C code is generated and compiled in a shared object and used in a driver code written in rust. The driver code prepare data, load the lib.so and profile it.

Issue: Little/no info available in the bottom-up view inside vtune-gui about the task involving a call to an external function defined in the dynamic library.

![image](https://github.com/intel/ittapi/assets/60969974/39e2fd3c-d3c4-4184-a040-94ebb3110919)

Doing exactly the same with a statically linked `extern fn` works as expected and gives performance info.

Is this a known issue ? Is there any workaround ?

Minimal reproducible project in the zip. 

[minimalissueittnotify.zip](https://github.com/intel/ittapi/files/11453237/minimalissueittnotify.zip)

**Comment by LXYan2333** [comment]
I'm not quite familier with rust, but did you try to compile the c module with debug  infomation? add "-g" flag in gcc/clang to compile c with debug info.

**Comment by ValouBambou** [comment]
I didn't remember exactly what I did last year, but my example wasn't using "-g" so it could be my bad. The only weird part I remember was that the static version and the dynamic version were both compiled without "-g" and only one of them can be analyzed in vtune was kind of weird. From what I tested with the current vtune version (on arch by time of writing 2024.1.0.596-3), this issue seems to have been solved in my minimal example at least.
